### PR TITLE
fix: Add multiple file paste support and fix for bugs

### LINKED ISSUE #4036: Add support for pasting multiple files from the clipboard
**Enhancement**

Now Chatwoot only allows pasting a single file from the clipboard even if we copy multiple files. So we want to add support for pasting multiple files from the clipboard.

**Bug**

1. When we copy a file from the finder and paste it into the normal editor if the input is active then the file name will show in input.
2. When we paste an image from the clipboard for attachment, if the advance editor input filed is active that image will show in the editor.

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

1. Add multiple files paste support.
2. Fixes showing file name in the editor field when we paste the file from finder.
2. Fixes showing the image in the advance editor when we paste the image as an attachment from the clipboard.

Fixes #4036 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/components/widgets/WootWriter/Editor.vue [hunk 1]
```diff
@@ -149,6 +154,18 @@ export default {
       this.$emit('toggle-canned-menu', !this.isPrivate && updatedValue);
     },
     value(newValue = '') {
+      if (this.hasImgTag && this.hasImgTag.length !== 0) {
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/WootWriter/Editor.vue:157]: This change will fix showing the image in the advance editor. @nithindavid @pranavrajs @fayazara Is there any other good for this issue?

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 2]
```diff
@@ -399,12 +399,19 @@ export default {
   methods: {
     onPaste(e) {
       const data = e.clipboardData.files;
+      if (!this.showRichContentEditor && data.length !== 0) {
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:402]: This change will solve the issue file name showing in the input field.

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 3]
```diff
@@ -372,6 +374,19 @@ export default {
       }
     },
     message(updatedMessage) {
+      if (this.hasFiles === true) {
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:377]: This change will fix showing the image in the advance editor. @nithindavid @pranavrajs @fayazara Is there any other good way to solve this issue?

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 4]
```diff
@@ -390,6 +392,19 @@ export default {
...
+        );
+        const imgElWrap = el.parentNode;
+        if (el) {
+          this.removeFocusEditorInputFileld();
+          this.$nextTick(() => {
+            imgElWrap.removeChild(el);
+          });
+        }
+        this.hasFiles = false;
+      }
```
Comment (pranavrajs) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:407]: @iamsivin This seems like a hack, can disable paste event for files in the prosemirror editor

**Review by nithindavid — APPROVED** [review]

**Comment by iamsivin** [comment]
@pranavrajs Can I merge this PR?

**Review by sojan-official — APPROVED** [review]

**Comment by iamsivin** [comment]
Ref: https://github.com/chatwoot/chatwoot/issues/4036#issuecomment-1076344178

**Pull request merged**